connectors/byoc.py

```diff
@@ -630,11 +638,16 @@ async def sync(self, elastic_server, idling, sync_now=False):
 
         try:
             service_type = self.service_type
-            if not sync_now:
+            if sync_now:
+                self.sync_now = True
+                logger.info("Sync forced")
+            else:
                 next_sync = self.next_sync()
                 if next_sync == SYNC_DISABLED or next_sync - idling > 0:
                     if next_sync == SYNC_DISABLED:
                         logger.debug(f"Scheduling is disabled for {service_type}")
+                    elif self.last_sync_status == JobStatus.SUSPENDED:
```

Comment:
This needs to be out of `if next_sync == SYNC_DISABLED or next_sync - idling > 0:` branch, or considering refactoring it to a new method as in:

https://github.com/elastic/connectors-python/blob/3e5c01202539bc8a3eac79f74d1b107242f9ae24/connectors/services/sync.py#L174-L188

`self.last_sync_status == JobStatus.SUSPENDED` can be added after `sync_now` check and before `next_sync = connector.next_sync()`

Reply:
Good catch!